From d310eaa56956e453a7d4a8b94787b5343d89902f Mon Sep 17 00:00:00 2001
From: Mahesh Kumar P <mahesh.kumar.p@intel.com>
Date: Tue, 7 Apr 2015 16:30:46 +0530
Subject: [PATCH] pm: Remove lpss resume delay for CHT

For CHT the resume delay can be removed. The same was removed on
Windows and also stress tested on Android. Removing this delay will
save ~200 ms of resume time.

Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-8116
Change-Id: I394afcf4bda28a5fc351557955a9a44129df526d
Signed-off-by: Kumar P, Mahesh <mahesh.kumar.p@intel.com>
---
 drivers/acpi/acpi_lpss.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/acpi/acpi_lpss.c b/drivers/acpi/acpi_lpss.c
index 956ad22..e380453 100644
--- a/drivers/acpi/acpi_lpss.c
+++ b/drivers/acpi/acpi_lpss.c
@@ -41,6 +41,8 @@ ACPI_MODULE_NAME("acpi_lpss");
 
 #define LPSS_PRV_REG_COUNT		9
 
+#define INTEL_ATOM_CHT 0x4c
+
 struct lpss_shared_clock {
 	const char *name;
 	unsigned long rate;
@@ -481,7 +483,8 @@ static void acpi_lpss_restore_ctx(struct device *dev)
 	 * expects 10ms delay before the device can be accessed after D3 to D0
 	 * transition.
 	 */
-	msleep(10);
+	if (!(boot_cpu_data.x86_model == INTEL_ATOM_CHT))
+		msleep(10);
 
 	for (i = 0; i < LPSS_PRV_REG_COUNT; i++) {
 		unsigned long offset = i * sizeof(u32);
-- 
1.9.1

